provisioning/baremetal_pod.go

```diff
@@ -818,7 +824,7 @@ func envWithProxy(proxy *configv1.Proxy, envVars []corev1.EnvVar) []corev1.EnvVa
 	return envVars
 }
 
-func newMetal3Deployment(info *ProvisioningInfo) *appsv1.Deployment {
+func newMetal3Deployment(info *ProvisioningInfo, sshKey string) *appsv1.Deployment {
```

Comment:
Does sshKey need to be a separate parameter when it is already part of ProvisioningInfo?